app/src/js/components/Input/edit.js

```diff
@@ -0,0 +1,165 @@
+    this.renderJson = this.renderJson.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+  }
+
+  async componentDidMount () {
+    const { inputId, controlId } = this.props.match.params;
+    await this.props.dispatch(getInput(inputId, controlId))
+  }
+
+  renderQuestionJson (name, data, refName) {
+    return (
+            <AceEditor
+                mode='json'
+                theme={config.editorTheme}
+                name={`edit-${name}`}
+                value={JSON.stringify(data, null, '\t')}
+                width='auto'
+                tabSize={config.tabSize}
+                showPrintMargin={false}
+                minLines={2}
+                maxLines={35}
+                wrapEnabled={true}
+                ref={refName}
+            />
+    );
+  }
+
+  async handleSubmit () {
+    const { dispatch } = this.props;
+    const { inputId, controlId } = this.props.match.params;
+    const inputAceEditorData = JSON.parse(this.refName.current.editor.getValue());
+
+    this.setState({ data: inputAceEditorData });
+
+    const stringifyFields = (fields) =>
+      Object.fromEntries(fields.map((field) => [field, JSON.stringify(inputAceEditorData[field])]));
+
+    const formattedPayload = {
+      ...inputAceEditorData,
+      edit_control_id: inputAceEditorData.control_id !== controlId? controlId: inputAceEditorData.control_id ,
+      ...stringifyFields(['enums', 'attributes', 'required_if', 'show_if'])
+    };
+
+    if (inputAceEditorData && Object.keys(inputAceEditorData).length !== 0) {
+      const result = await dispatch(inputId ? updateInput(formattedPayload) : createInput(formattedPayload));
+      if(!result?.statusCode) this.props.history.push(`/inputs/edit/${inputAceEditorData?.question_id}/${formattedPayload?.control_id}`);
+      if(inputId) this.props.history.push('/inputs');
+    }    
+  }
+
+  getRandom () {
+    return Math.random().toString(36).substring(2, 15) + Math.random().toString(36).substring(2, 15);
+  }
+
+  render () {
+    const { inputId } = this.props.match.params;
+
+    const record = (inputId ? this.props.questions.detail : { data: {} });
+    const { canCreate, canEdit } = formPrivilegesCU(this.props.privileges);
+    const breadcrumbConfig = [
+      {
+        label: 'Dashboard Home',
+        href: '/'
+      },
+      {
+        label: 'Inputs',
+        href: '/inputs'
+      },
+      {
+        label: inputId || 'Add Input',
+        active: true
+      }
+    ];
+    console.log('record',record)
+    return (
+            <div className='page__component'>
+                <section className='page__section page__section__controls'>
+                    <Breadcrumbs config={breadcrumbConfig} />
+                </section>
+                <h1 className='heading--large heading--shared-content with-description'>
+                    {record.data ? (record.data.control_id ?  'Edit Input' : 'Add Input') : '...'}
```

Comment:
Not changing after page re-renders on successful addition

Reply:
After successful addition, it goes to the Edit page with the current code structure. I will redirect it to /inputs.